File: Core kernel Yaiwes/Backend watchdog workflow adaptativo/sandbox-runtime/iii-sandbox/packages/sdk-rust/tests/filesystem_test.rs
use std::sync::Arc;

use iii_sandbox::filesystem::FileSystem;
use iii_sandbox::{ClientConfig, HttpClient};

fn make_fs(url: &str) -> FileSystem {
    let client = Arc::new(HttpClient::new(ClientConfig {
        base_url: url.to_string(),
        token: None,
        timeout_ms: None,
    }).unwrap());
    FileSystem::new(client, "sbx-1".into())
}

#[tokio::test]
async fn test_read() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/read")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(serde_json::json!("file contents here").to_string())
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    let content = fs.read("/workspace/main.py").await.unwrap();
    assert_eq!(content, "file contents here");
    mock.assert_async().await;
}

#[tokio::test]
async fn test_write() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/write")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body("{}")
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    fs.write("/workspace/main.py", "print('hello')").await.unwrap();
    mock.assert_async().await;
}

#[tokio::test]
async fn test_delete() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/delete")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body("{}")
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    fs.delete("/workspace/old.py").await.unwrap();
    mock.assert_async().await;
}

#[tokio::test]
async fn test_list_default_path() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/list")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(
            serde_json::json!([
                {
                    "name": "main.py",
                    "path": "/workspace/main.py",
                    "size": 100,
                    "isDirectory": false,
                    "modifiedAt": 1000
                },
                {
                    "name": "src",
                    "path": "/workspace/src",
                    "size": 0,
                    "isDirectory": true,
                    "modifiedAt": 2000
                }
            ])
            .to_string(),
        )
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    let files = fs.list(None).await.unwrap();
    assert_eq!(files.len(), 2);
    assert_eq!(files[0].name, "main.py");
    assert!(!files[0].is_directory);
    assert!(files[1].is_directory);
    mock.assert_async().await;
}

#[tokio::test]
async fn test_list_custom_path() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/list")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(
            serde_json::json!([
                {
                    "name": "lib.rs",
                    "path": "/workspace/src/lib.rs",
                    "size": 200,
                    "isDirectory": false,
                    "modifiedAt": 3000
                }
            ])
            .to_string(),
        )
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    let files = fs.list(Some("/workspace/src")).await.unwrap();
    assert_eq!(files.len(), 1);
    assert_eq!(files[0].name, "lib.rs");
    mock.assert_async().await;
}

#[tokio::test]
async fn test_search() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/search")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(
            serde_json::json!([
                "/workspace/main.py",
                "/workspace/test.py"
            ])
            .to_string(),
        )
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    let results = fs.search("*.py", None).await.unwrap();
    assert_eq!(results.len(), 2);
    assert_eq!(results[0], "/workspace/main.py");
    mock.assert_async().await;
}

#[tokio::test]
async fn test_upload() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/upload")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body("{}")
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    fs.upload("/workspace/data.csv", "a,b,c\n1,2,3")
        .await
        .unwrap();
    mock.assert_async().await;
}

#[tokio::test]
async fn test_download() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/files/download")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(serde_json::json!("downloaded content").to_string())
        .create_async()
        .await;

    let fs = make_fs(&server.url());
    let content = fs.download("/workspace/data.csv").await.unwrap();
    assert_eq!(content, "downloaded content");
    mock.assert_async().await;
}

File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/test/sdk/filesystem.test.ts
import { describe, it, expect, vi, beforeEach } from "vitest"
import { FileSystem } from "../../packages/sdk/src/filesystem.js"
import { HttpClient } from "../../packages/sdk/src/client.js"

describe("FileSystem", () => {
  let mockClient: HttpClient
  let fs: FileSystem

  beforeEach(() => {
    mockClient = {
      get: vi.fn(),
      post: vi.fn(),
      del: vi.fn(),
      stream: vi.fn(),
    } as unknown as HttpClient
    fs = new FileSystem(mockClient, "sbx_test")
  })

  describe("read", () => {
    it("calls correct endpoint with path", async () => {
      ;(mockClient.post as any).mockResolvedValue("file content")
      const result = await fs.read("/workspace/test.py")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/read",
        { path: "/workspace/test.py" },
      )
      expect(result).toBe("file content")
    })
  })

  describe("write", () => {
    it("calls correct endpoint with path and content", async () => {
      ;(mockClient.post as any).mockResolvedValue(undefined)
      await fs.write("/workspace/out.txt", "hello world")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/write",
        { path: "/workspace/out.txt", content: "hello world" },
      )
    })
  })

  describe("delete", () => {
    it("calls correct endpoint", async () => {
      ;(mockClient.post as any).mockResolvedValue(undefined)
      await fs.delete("/workspace/old.txt")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/delete",
        { path: "/workspace/old.txt" },
      )
    })
  })

  describe("list", () => {
    it("uses default path /workspace", async () => {
      const files = [{ name: "a.py", path: "/workspace/a.py", size: 100, isDirectory: false, modifiedAt: 1 }]
      ;(mockClient.post as any).mockResolvedValue(files)
      const result = await fs.list()

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/list",
        { path: "/workspace" },
      )
      expect(result).toEqual(files)
    })

    it("uses custom path when provided", async () => {
      ;(mockClient.post as any).mockResolvedValue([])
      await fs.list("/workspace/src")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/list",
        { path: "/workspace/src" },
      )
    })
  })

  describe("search", () => {
    it("uses default dir /workspace", async () => {
      ;(mockClient.post as any).mockResolvedValue(["/workspace/a.py"])
      await fs.search("*.py")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/search",
        { pattern: "*.py", dir: "/workspace" },
      )
    })

    it("uses custom dir when provided", async () => {
      ;(mockClient.post as any).mockResolvedValue([])
      await fs.search("*.ts", "/workspace/src")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/search",
        { pattern: "*.ts", dir: "/workspace/src" },
      )
    })
  })

  describe("upload", () => {
    it("sends base64 content", async () => {
      ;(mockClient.post as any).mockResolvedValue(undefined)
      await fs.upload("/workspace/data.bin", "aGVsbG8=")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/upload",
        { path: "/workspace/data.bin", content: "aGVsbG8=" },
      )
    })
  })

  describe("download", () => {
    it("returns base64 content", async () => {
      ;(mockClient.post as any).mockResolvedValue("aGVsbG8=")
      const result = await fs.download("/workspace/data.bin")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/files/download",
        { path: "/workspace/data.bin" },
      )
      expect(result).toBe("aGVsbG8=")
    })
  })
})
